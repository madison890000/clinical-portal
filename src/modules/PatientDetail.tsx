import Name from '../components/Name';
import { Card, CardContent, Skeleton } from '@mui/material';
import InfoItem from '../components/InfoItem';
import styles from './PatientDetail.module.scss';
import { useQuery } from 'react-query';
import { getPatientById } from '../services';
import React from 'react';

const PatientDetail = ({ id }: { id: string }) => {
    const { data: patient } = useQuery(['getPatientDetail', id], () => getPatientById(id));

    return (
        <>
            {!patient && <Skeleton variant="rectangular" width={800} height={300} />}
            {patient && (
                <Card className={styles.container}>
                    <CardContent>
                        <Name {...patient} />
                    </CardContent>
                    <CardContent>
                        <InfoItem name="sex" value={patient?.sex} />
                        <InfoItem name="age" value={patient?.age} />
                    </CardContent>
                </Card>
            )}
        </>
    );
};

export default PatientDetail;
