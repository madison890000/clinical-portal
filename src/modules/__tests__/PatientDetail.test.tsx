import React from 'react';
import PatientDetail from '../PatientDetail';
import { render, waitFor } from '@testing-library/react';

jest.mock('../../services', () => {
    return {
        getPatientById: () => {
            return Promise.resolve({
                title: 'Mr.',
                firstName: 'James',
                preferredName: 'Jay',
                middleName: 'Shaun',
                familyName: 'Cooper',
                age: 17,
                sex: 'Indeterminate'
            });
        }
    };
});
test('renders PatientDetail correctly', async () => {
    const { container } = render(<PatientDetail id="test" />);
    await waitFor(() => {
        expect(container).toMatchSnapshot();
    });
});
