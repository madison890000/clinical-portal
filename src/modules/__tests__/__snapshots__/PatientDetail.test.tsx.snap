// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PatientDetail correctly 1`] = `
<div>
  <span
    class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
    style="width: 800px; height: 300px;"
  />
</div>
`;
